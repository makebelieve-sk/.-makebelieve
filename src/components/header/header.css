.game_header {
    display: flex;
    justify-content: space-around;
    align-items: center;
    width: 100%; 
    margin: 0;
    position: absolute;
    top: 0;
    left: 0;
}
.wrapper-logo {
    display: flex;
}
.button-back {
    margin-top: 23px;
    margin-right: 10px;
    align-items: center;
    width: 51px;
    height: 40px;
    border: none;
    transform: rotate(180deg);
    background: url('./img/button-back.png');
    background-repeat: no-repeat;
}
.button-back:hover {
    cursor: pointer;
}
.welcome_logo_header {
    width: 177px;
    height: 76px;
    margin-top: 5px;
    background: url('./img/melody-logo-ginger.png');
    background-repeat: no-repeat;
}

.welcome_logo_header:hover {
    cursor: pointer;
}