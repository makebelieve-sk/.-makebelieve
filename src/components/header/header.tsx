import React from 'react';

import Timer from '../timer/timer';
import { Mistakes } from '../mistakes/mistakes';
import './header.css';

type HeaderType = {
    onBack: () => any,
    onHome: () => any,
    mistakes: number,
    timeEnd: () => any,
    getWinTime: (min: string, sec: string) => any
};

export const Header: React.FC<HeaderType> = ({ onBack, onHome, mistakes, timeEnd, getWinTime }) => {
    return (
        <header className="game_header">
            <div className="wrapper-logo">
                <button className="button-back" onClick={() => onBack()}/>
                <i className="welcome_logo_header" onClick={() => onHome()}/>
            </div>
            <Timer timeEnd={timeEnd} getWinTime={getWinTime} />
            <Mistakes mistakes={mistakes}/>
        </header>
    );
};