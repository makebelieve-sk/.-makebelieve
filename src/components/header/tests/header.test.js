import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Header } from '../header';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3, prop4, prop5) => shallow(
    <Header onBack={prop1} onHome={prop2} mistakes={prop3} timeEnd={prop4} getWinTime={prop5} />
);

const mock = {
    onBack: jest.fn(),
    onHome: jest.fn(),
    mistakes: 1,
    timeEnd: jest.fn(),
    getWinTime: jest.fn()
};

describe(`Header component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.onBack, mock.onHome, mock.mistakes, mock.timeEnd, mock.getWinTime);
    });

    it(`should contain header wrapper`, () => {
        const wrapper = component.find('.game_header');
        expect(wrapper.length).toBe(1);
    });

    it(`Header correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<Header 
                onBack={mock.onBack}
                onHome={mock.onHome}
                mistakes={mock.mistakes}
                timeEnd={mock.timeEnd}
                getWinTime={mock.getWinTime}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});