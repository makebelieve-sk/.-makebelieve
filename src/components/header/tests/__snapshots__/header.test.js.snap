// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header correctly renders after relaunch 1`] = `
<header
  className="game_header"
>
  <div
    className="wrapper-logo"
  >
    <button
      className="button-back"
      onClick={[Function]}
    />
    <i
      className="welcome_logo_header"
      onClick={[Function]}
    />
  </div>
  <div
    className="timer_value"
  >
    <span
      className="timer_mins"
    >
      05
    </span>
    <span
      className="timer_dots"
    >
      :
    </span>
    <span
      className="timer_secs"
    >
      00
    </span>
  </div>
  <div
    className="game_mistakes"
  >
    <div
      className="wrong"
    />
    <div
      className="wrong"
    />
    <div
      className="wrong-active"
    />
  </div>
</header>
`;
